lib/hoe/test.rb

```diff
@@ -80,8 +80,9 @@ def define_test_tasks
       when :minitest then
         require "minitest/test_task" # currently in hoe, but will move
 
+        test_prelude = self.test_prelude
         Minitest::TestTask.create :test do |t|
-          t.test_prelude = self.test_prelude
+          t.test_prelude = test_prelude
```

Comment:
I'm not sure how to test this one w/o assigning the created task to an ivar (which I'm not against).

Reply:
I'm now pretty sure you _can't_ test this properly without really digging through rake internals. My tests are only really testing `make_test_command` and some of the initializers, but the method `define_test_tasks` is pretty much untouched.

Reply:
:thinking:

We could write a test like this to shell out and examine the output of the `test:cmd` task:

```ruby
class TestTestTasks < Minitest::Test
  def test_define_test_task_for_minitest
    pwd = File.expand_path(File.join(File.dirname(__FILE__), ".."))

    rakefile = <<~EORAKE
      # ensure the version of hoe under test is loaded
      $:.prepend "#{pwd}/lib"
      require "hoe"

      Hoe.spec "fakegem" do
        # minimum config necessary
        developer "you", "you@example.com"

        # avoid warning messages
        license "MIT"

        # perhaps unnecessary given it's the default, but let's be explicit
        self.testlib = :minitest

        # this is the bit we're testing for
        self.test_prelude = 'require "other/file"'
      end
    EORAKE

    command = Tempfile.open("hoe_test_rakefile") do |f|
      f.write rakefile
      f.close
      `rake -f #{f.path} test:cmd`
    end

    assert_match %r{require "other/file"; require "minitest/autorun"}, command
  end
end
```

... but this seems a bit brittle.

Reply:
Yeah. Shelling out feels too brittle for too little benefit. I can refactor the task creation code out to make it more directly testable at the very least. Either way... this code has been fixed.